src/ValidatorKeys.cpp

```diff
@@ -0,0 +1,191 @@
+*/
+//==============================================================================
+
+#include <ValidatorKeys.h>
+#include <ripple/json/json_reader.h>
+#include <ripple/protocol/HashPrefix.h>
+#include <ripple/protocol/Sign.h>
+#include <beast/core/detail/base64.hpp>
+#include <boost/filesystem.hpp>
+#include <fstream>
+
+namespace ripple {
+
+ValidatorKeys::ValidatorKeys (KeyType const& keyType)
+    : keyType_ (keyType)
+    , sequence_ (0)
+{
+    auto seed = randomSeed ();
+    auto const kp = generateKeyPair (keyType_, seed);
+
+    masterSecret_ = kp.second;
+    masterPublicKey_ = kp.first;
+}
+
+ValidatorKeys::ValidatorKeys (
+    KeyType const& keyType,
+    PublicKey const& masterPublicKey,
+    SecretKey const& masterSecret,
+    std::uint32_t sequence)
+    : keyType_ (keyType)
+    , masterPublicKey_ (masterPublicKey)
+    , masterSecret_ (masterSecret)
+    , sequence_ (sequence)
+{
+}
+
+ValidatorKeys
+ValidatorKeys::make_ValidatorKeys (
+    boost::filesystem::path const& keyFile)
+{
+    std::ifstream ifsKeys (keyFile.c_str (), std::ios::in);
+
+    if (! ifsKeys)
+        throw std::runtime_error (
+            "Failed to open key file: " + keyFile.string());
+
+    Json::Reader reader;
+    Json::Value jKeys;
+    if (! reader.parse (ifsKeys, jKeys))
+    {
+        throw std::runtime_error (
+            "Unable to parse json key file: " + keyFile.string());
+    }
+
+    static std::array<std::string, 3> const requiredFields {{
+        "key_type",
+        "master_secret_key",
+        "sequence"
+    }};
+
+    for (auto field : requiredFields)
+    {
+        if (! jKeys.isMember(field))
+        {
+            throw std::runtime_error (
+                "Key file '" + keyFile.string() +
+                "' is missing \"" + field + "\" field");
+        }
+    }
+
+    auto const keyType = keyTypeFromString (jKeys["key_type"].asString());
+    if (keyType == KeyType::invalid)
+    {
+        throw std::runtime_error (
+            "Key file '" + keyFile.string() +
+            "' contains invalid key type: " +
+            jKeys["key_type"].toStyledString());
+    }
+
+    auto const masterSecret = parseBase58<SecretKey> (
+        TokenType::TOKEN_NODE_PRIVATE, jKeys["master_secret_key"].asString());
+
+    if (! masterSecret)
+    {
+        throw std::runtime_error (
+            "Key file '" + keyFile.string() +
+            "' contains invalid master secret: " +
+            jKeys["master_secret_key"].toStyledString());
+    }
+
+    auto const masterPublicKey = derivePublicKey(keyType, *masterSecret);
+
+    if (! jKeys["sequence"].isIntegral())
+    {
+        throw std::runtime_error (
+            "Key file '" + keyFile.string() +
+            "' contains invalid sequence: " +
+            jKeys["sequence"].toStyledString());
+    }
+
+    std::uint32_t sequence;
+    try {
+        sequence = jKeys["sequence"].asUInt();
+    }
+    catch (std::runtime_error& e)
```

Comment:
Variable `e` is unused. It looks like you don't want `e.what()` in the error message, so you can just catch without it: `catch (std::runtime_error&)`.